MODULE.bazel

```diff
@@ -1,6 +1,49 @@
-###############################################################################
-# Bazel now uses Bzlmod by default to manage external dependencies.
-# Please consider migrating your external dependencies from WORKSPACE to MODULE.bazel.
-#
-# For more details, please check https://github.com/bazelbuild/bazel/issues/18958
-###############################################################################
+module(
+    name = "grpc_ecosystem_grpc_gateway",
+    version = "2.23.0",
+)
+
+# Bazel Central Registry modules.
+bazel_dep(name = "bazel_features", version = "1.20.0")
+bazel_dep(name = "rules_license", version = "1.0.0")
+bazel_dep(name = "rules_python", version = "0.37.2")
+bazel_dep(name = "rules_proto", version = "6.0.2")
+bazel_dep(name = "rules_go", version = "0.50.1", repo_name = "io_bazel_rules_go")
+bazel_dep(name = "gazelle", version = "0.39.1", repo_name = "bazel_gazelle")
+bazel_dep(name = "bazel_skylib", version = "1.7.1")
+bazel_dep(name = "protobuf", version = "26.0.bcr.2", repo_name = "com_google_protobuf")
+bazel_dep(name = "googleapis", version = "0.0.0-20240819-fe8ba054a")
+
+go_sdk = use_extension("@io_bazel_rules_go//go:extensions.bzl", "go_sdk")
+go_sdk.download(version = "1.22.7")
+
+go_deps = use_extension("@bazel_gazelle//:extensions.bzl", "go_deps")
+go_deps.from_file(go_mod = ":go.mod")
+go_deps.module(
+    path = "google.golang.org/grpc/cmd/protoc-gen-go-grpc",
+    sum = "h1:rNBFJjBCOgVr9pWD7rs/knKL4FRTKgpZmsRfV214zcA=",
+    version = "v1.3.0",
+)
+go_deps.module(
+    path = "github.com/golang/protobuf",
+    sum = "h1:i7eJL8qZTpSEXOPTxNKhASYpMn+8e5Q6AdndVa1dWek=",
+    version = "v1.5.4",
+)
+use_repo(
+    go_deps,
+    "com_github_antihax_optional",
+    "com_github_google_go_cmp",
+    "com_github_golang_protobuf",
+    "com_github_rogpeppe_fastuuid",
+    "in_gopkg_yaml_v3",
+    "org_golang_google_genproto_googleapis_api",
+    "org_golang_google_genproto_googleapis_rpc",
+    "org_golang_google_grpc",
+    "org_golang_google_grpc_cmd_protoc_gen_go_grpc",
+    "org_golang_google_protobuf",
+    "org_golang_x_oauth2",
+    "org_golang_x_text",
```

Comment:
That's right. This has to do with the way Bazel modules and Gazelle work that is a little bit different from working with Bazel workspaces.

This was sort of already happening in the same way with the existing WORKSPACE workflow, only the dependencies were defined in the [`repositories.bzl`](https://github.com/grpc-ecosystem/grpc-gateway/blob/main/repositories.bzl) file and [loaded in the WORKSPACE instead](https://github.com/grpc-ecosystem/grpc-gateway/blob/main/WORKSPACE#L117-L119).

With this new workflow and because of some changes in how Bazel modules work with respect to workspaces, we need to make these explicitly available to the project with `use_repo`.

> Will buildifier update this file if we add new dependencies to go.mod?

Yes, whenever we run `bazel run //:gazelle`, Gazelle will detect differences and prompt the dev to run `buildifier` to update this file.